src/main/java/org/candlepin/service/model/EventBody.java

```diff
@@ -0,0 +1,114 @@
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.service.model;
+
+import java.util.Objects;
+
+/**
+ * The event body contains the serialized text for the {@link Event}'s body and what type of serialization
+ * was used to create the text.
+ */
+public class EventBody {
+
+    private String body;
+    private SerializationType serializationType;
+
+    /**
+     * Creates a new event body with the provided body and serialization type.
+     *
+     * @param body
+     *  the serialized body of an {@link Event}
+     *
+     * @param serializationType
+     *  the type of serialization used to create the body text
+     *
+     * @throws IllegalArgumentException
+     *  if the provided body text is null or blank
+     *
+     * @throws NullPointerException
+     *  if the provided serialization type is null
+     */
+    public EventBody(String body, SerializationType serializationType) {
+        if (body == null) {
+            throw new IllegalArgumentException("event body is null");
+        }
+
+        if (body.isBlank()) {
+            throw new IllegalArgumentException("event body is blank");
+        }
+
+        this.body = body;
+        this.serializationType = Objects.requireNonNull(serializationType);
+    }
+
+    /**
+     * Sets the serialized body text and overrides any existing value that has already been set.
+     *
+     * @param body
+     *  the serialized body text to set
+     *
+     * @throws IllegalArgumentException
+     *  if the provided body text is null or blank
+     *
+     * @return this instance
+     */
+    public EventBody setBody(String body) {
+        if (body == null) {
+            throw new IllegalArgumentException("event body is null");
+        }
+
+        if (body.isBlank()) {
+            throw new IllegalArgumentException("event body is blank");
+        }
+
+        this.body = body;
+        return this;
+    }
+
+    /**
+     * @return the serialized {@link Event} body
+     */
+    public String getBody() {
+        return body;
+    }
+
+    /**
+     * Sets the serialization type used to create the body text.
+     *
+     * @param type
+     *  the serialization type used to create the body text
+     *
+     * @throws IllegalArgumentException
+     *  if the provided serialization type is null
+     *
+     * @return this instance
+     */
+    public EventBody setSerializationType(SerializationType type) {
+        if (type  == null) {
+            throw new IllegalArgumentException("serialization type is null");
+        }
+
+        this.serializationType = type;
+        return this;
+    }
+
+    /**
+     * @return the type of serialization used to create the body text
+     */
+    public SerializationType getSerializationType() {
```

Comment:
The serialization type looks to be a function of the underlying event -- either the specific event itself, or the eventing platform/interface. It doesn't look like something that will be changing arbitrarily to need this kind of behavior.

What was the intent behind separating the event body into the (string, type enum) pair for every event over something like having both be available on the event itself or declaring the format as part of the event API?

Reply:
The intention was being able to enforce that the serialized event string will not be null or blank and the serialization type will not be null because of the validation done in the EventBody class.  We could have the serialized body string and the serialization type be returned by the event interface and have java docs explaining that the values should not be null/blank. 

I don't feel too strongly either way, but that was my thought process on breaking the EventBody into another class.

Reply:
After finally revisiting this, I think it makes more sense to just have the `getBody()` and `getSerializationType()` methods on the `Event` interface instead of having an `EventBody` object.  Updated